packages/drivers/routerlicious-driver/src/test/r11sSocketTests.spec.ts

```diff
@@ -69,129 +74,191 @@ describe("R11s Socket Tests", () => {
+		clientError: any,
+		expectedErrorType: string,
+	) {
+		it(description, async () => {
+			const socketEventName = "connect_document_success";
+			socket = new ClientSocketMock({
+				connect_document: { eventToEmit: socketEventName },
+			});
+			const connection = await mockSocket(socket as unknown as Socket, async () =>
+				documentService.connectToDeltaStream(client),
+			);
+			const disconnectEventP = new Promise<{ clientId: string; errorType: string }>(
+				(resolve) => {
+					assert(socket !== undefined, "Socket should be defined");
+					socket.on(
+						"disconnect_document",
+						(clientId: string, _documentId: string, errorType: string) => {
+							resolve({ clientId, errorType });
+						},
+					);
+				},
+			);
+			(connection as any).disconnect(clientError);
+			const disconnectResult = await disconnectEventP;
+			assert.strictEqual(
+				disconnectResult.clientId,
+				connection.clientId,
+				"Client ID should match",
+			);
+			assert.strictEqual(
+				disconnectResult.errorType,
+				expectedErrorType,
+				`Error type should be ${expectedErrorType}`,
+			);
 		});
+	}
 
-		// Send Token Revoked error
-		socket.sendErrorEvent(errorToThrow);
-
-		assert(
-			error?.errorType === DriverErrorTypes.authorizationError,
-			"Error type should be authorizationError",
-		);
-		assert(error.scenarioName === "error", "Error scenario name should be error");
-		assert(
-			(error as any).internalErrorCode === "TokenRevoked",
-			"Error internal code should be TokenRevoked",
-		);
-	});
+	// connect_document_error tests
+	const connectErrorTests = [
+		{
+			description: "connect_document_error with Token Revoked error",
+			errorToThrow: {
+				code: 403,
+				message: "TokenRevokedError",
+				retryAfterMs: 10,
+				internalErrorCode: "TokenRevoked",
+				errorType: DriverErrorTypes.authorizationError,
+				canRetry: false,
+			},
+			expectedErrorType: DriverErrorTypes.authorizationError,
+			expectedInternalErrorCode: "TokenRevoked",
+		},
+		{
+			description: "connect_document_error with Cluster Draining error",
+			errorToThrow: {
+				code: 503,
+				message: "ClusterDrainingError",
+				retryAfterMs: 1000,
+				internalErrorCode: R11sServiceClusterDrainingErrorCode,
+				errorType: RouterliciousErrorTypes.clusterDrainingError,
+				canRetry: true,
+			},
+			expectedErrorType: RouterliciousErrorTypes.clusterDrainingError,
+			expectedInternalErrorCode: R11sServiceClusterDrainingErrorCode,
+		},
+	];
 
-	it("connect_document_error with Cluster Draining error", async () => {
-		const errorToThrow = {
-			code: 503,
-			message: "ClusterDrainingError",
-			retryAfterMs: 1000,
-			internalErrorCode: R11sServiceClusterDrainingErrorCode,
-			errorType: RouterliciousErrorTypes.clusterDrainingError,
-			canRetry: true,
-		};
-		const errorEventName = "connect_document_error";
-		socket = new ClientSocketMock({
-			connect_document: { eventToEmit: errorEventName, errorToThrow },
-		});
+	connectErrorTests.forEach((test) =>
```

Comment:
Nit: use `for ... of ...`. Our linter doesn't like `forEach`, but this package is still using the minimal-deprecated lint config so it doesn't complain.